.github/workflows/snyk.yml

```diff
@@ -0,0 +1,66 @@
+---
+name: Check for Snyk Vulnerabilities
+
+on:   # yamllint disable-line rule:truthy
+  workflow_dispatch:
+  # schedule:
+  #   - cron: '0 12 * * *'  # every day at 12pm UTC
+
+jobs:
+  snyk:
+    name: snyk test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install Dependencies
+        run: >
+          npm install snyk -g;
+          pip install -r requirements.txt;
+      - name: Run Snyk Scan
+        run: >
+          # Run scan
+          snyk test --file=requirements.txt --json-file-output=scan.json;
+
+          # Exit if no vulnerabilities
+          [[ "$(jq '.ok' scan.json)" == "true" ]] && exit 0;
+          while read -r fix
+          do
+            package=$(echo $fix | cut -d "=" -f 1)
+            sed -i "/${package}.*/d" ckan/requirements.in
+            sed -i "/${package}.*/d" ckan/requirements.txt
+            echo $fix >> ckan/requirements.in
+          done <<< $(jq -r '.vulnerabilities[] | .moduleName,.fixedIn[]'
+            scan.json | sed 'N;s/\n/>=/');
+
+          make update-dependencies;
+          exit 1
+      - name: Create Pull Request
+        if: ${{ failure() }}
```

Comment:
Shouldn't we want to do this on success, not failure?

Reply:
The previous command succeeds if there are no vulnerabilities and fails if there are vulnerabilities.